### PR TITLE
Faucet v2

### PR DESCRIPTION
**Pull request by pitasi**
A web-based faucet that uses recaptcha v3 for validating incoming requests.

I modified spaceward "Get WARD" buttons to open the faucet url in a new tab instead of making the HTTP request in the background.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new web-based version of the faucet with reCAPTCHA verification for enhanced security.
  - Added rate-limiting functionality to manage request frequency.
  - Implemented Prometheus metrics for tracking faucet service statistics.

- **Bug Fixes**
  - Updated button behavior to open a URL in a new tab on click for better user experience.

- **Chores**
  - Updated `.gitignore` to ignore `.envrc` files.
  - Revised Dockerfile to transition to version 2 of the faucet component.
  - Updated GitHub workflows to include paths and configurations for the new faucet-v2 service.

### CODE REVIEW

File: cmd/faucet-v2/client.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		"q",
+		"tx",
+		txHash,
+		"--node",
+		c.cfg.Node,
+		"-o",
+		"json",
+	}, " ")
+
+	deadline, cancel := context.WithTimeout(ctx, 10*time.Second)
```
Comment (jlehtimaki) [inline cmd/faucet-v2/client.go]: I think the 10 should be a const?

File: cmd/faucet-v2/main.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		AddressLimiter: NewLimiter(c.Cooldown),
+		IPLimiter:      NewLimiter(c.Cooldown),
+		Client:         client,
+	}
+
+	http.HandleFunc("/", formHandler(c))
+	http.HandleFunc("/submit", submitHandler(c, env))
+	http.Handle("/metrics", promhttp.Handler())
+	logger.Info("faucet listening at :8000")
+	if err := http.ListenAndServe(":8000", nil); err != nil {
```
Comment (jlehtimaki) [inline cmd/faucet-v2/main.go]: I think it would be good to have a timeout here?

File: cmd/faucet-v2/metrics.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		Help:      "The size of the batch of addresses waiting for tokens",
+	})
+
+	reqCount = promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: "faucet",
+		Name:      "req_total",
+		Help:      "The total number of faucet requests",
+	})
+
+	reqInvalidIpCount = promauto.NewCounter(prometheus.CounterOpts{
```
Comment (jlehtimaki) [inline cmd/faucet-v2/metrics.go]:
```suggestion
	reqInvalidIPCount = promauto.NewCounter(prometheus.CounterOpts{
```

File: cmd/faucet-v2/metrics.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	reqInvalidCaptchaCount = promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: "faucet",
+		Name:      "req_invalid_captcha_total",
+		Help:      "The total number of failed requests for invalid captcha",
+	})
+
+	reqErrorCount = promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: "faucet",
+		Name:      "req_error_count",
```
Comment (jlehtimaki) [inline cmd/faucet-v2/metrics.go:47]: I think usually these are in Prometheus with `_total` suffix

File: cmd/faucet-v2/main.go (hunk 2 above)
Comment (pitasi) [reply cmd/faucet-v2/main.go]: sorry, where?

File: cmd/faucet-v2/metrics.go (hunk 3 above)
Comment (pitasi) [reply cmd/faucet-v2/metrics.go]: done

File: cmd/faucet-v2/metrics.go (hunk 4 above)
Comment (pitasi) [reply cmd/faucet-v2/metrics.go:47]: done!

File: cmd/faucet-v2/client.go (hunk 1 above)
Comment (pitasi) [reply cmd/faucet-v2/client.go]: moved into a config variable, and increased to 15 seconds so we're sure it waits for at least a couple blocks

File: cmd/faucet-v2/main.go (hunk 2 above)
Comment (jlehtimaki) [reply cmd/faucet-v2/main.go]: So that the clients cannot uphold a connection indefenetively

Comment (pitasi) [reply cmd/faucet-v2/main.go]: I added `ReadHeaderTimeout`, is this enough or should I set more timeouts?

Comment (jlehtimaki) [reply cmd/faucet-v2/main.go]: lgtm

**Review by jlehtimaki — APPROVED** [review]

**Pull request merged**